addons/note/note.py

```diff
@@ -20,125 +20,116 @@
 
 from openerp import SUPERUSER_ID
-from openerp.osv import osv, fields
+import openerp
+from openerp import models, fields, api, exceptions, _
 from openerp.tools import html2plaintext
+from datetime import datetime
+from openerp.tools import DEFAULT_SERVER_DATETIME_FORMAT
 
-class note_stage(osv.osv):
+class note_stage(models.Model):
     """ Category of Note """
     _name = "note.stage"
     _description = "Note Stage"
-    _columns = {
-        'name': fields.char('Stage Name', translate=True, required=True),
-        'sequence': fields.integer('Sequence', help="Used to order the note stages"),
-        'user_id': fields.many2one('res.users', 'Owner', help="Owner of the note stage.", required=True, ondelete='cascade'),
-        'fold': fields.boolean('Folded by Default'),
-    }
+
+    name = fields.Char('Stage Name', translate=True, required=True)
+    sequence = fields.Integer('Sequence', help="Used to order the note stages", default=1)
+    user_id = fields.Many2one('res.users', 'Owner', help="Owner of the note stage.", required=True, ondelete='cascade', default=lambda self: self.env.user)
+    fold = fields.Boolean('Folded by Default', default=0)
+
     _order = 'sequence asc'
-    _defaults = {
-        'fold': 0,
-        'user_id': lambda self, cr, uid, ctx: uid,
-        'sequence' : 1,
-    }
 
-class note_tag(osv.osv):
+class note_tag(models.Model):
     _name = "note.tag"
     _description = "Note Tag"
-    _columns = {
-        'name' : fields.char('Tag Name', required=True),
-    }
+    name = fields.Char('Tag Name', required=True)
 
-class note_note(osv.osv):
+class note_note(models.Model):
     """ Note """
     _name = 'note.note'
     _inherit = ['mail.thread']
     _description = "Note"
 
     #writing method (no modification of values)
-    def name_create(self, cr, uid, name, context=None):
-        rec_id = self.create(cr, uid, {'memo': name}, context=context)
-        return self.name_get(cr, uid, [rec_id], context)[0]
+    #overwrite models.Model.name_create(): creates a new record with only the name provided
+    @api.model
+    def name_create(self, name):
+        stage_id = self.env.context['default_stage_id']
+        rec_id = self.create({'memo': name, 'stage_ids':[(6,0,[stage_id])]})
+        return rec_id.name_get()[0]
 
     #read the first line (convert hml into text)
-    def _get_note_first_line(self, cr, uid, ids, name="", args={}, context=None):
-        res = {}
-        for note in self.browse(cr, uid, ids, context=context):
-            res[note.id] = (note.memo and html2plaintext(note.memo) or "").strip().replace('*','').split("\n")[0]
+    @api.one
+    @api.depends('memo')
+    def _get_note_first_line(self):
+        self.name = (self.memo and html2plaintext(self.memo) or "").strip().replace('*','').split("\n")[0]
 
-        return res
+    @api.one
+    def onclick_note_is_done(self):
+        self.open = 0
+        self.date_done = datetime.now().strftime(DEFAULT_SERVER_DATETIME_FORMAT)
```

Comment:
You certainly want to use `fields.Datetime` helpers instead of `DEFAULT_SERVER_DATETIME_FORMAT`